zoo/policies/__init__.py

```diff
@@ -100,3 +108,126 @@ def human_keyboard_entrypoint(*arg, **kwargs):
+    )  # folder contains all competition environment
+    sub_env_path = os.path.join(
+        comp_env_path, f"{Path(policy_path).name}"
+    )  # folder contains single competition environment
+    req_file = os.path.join(
+        policy_path, "requirements.txt"
+    )  # path of the requiremnet file
+
+    if Path(sub_env_path).exists():
+        shutil.rmtree(sub_env_path)
+    Path.mkdir(Path(sub_env_path), parents=True, exist_ok=True)
+
+    try:
+        subprocess.check_call(
+            [
+                sys.executable,
+                "-m",
+                "pip",
+                "install",
+                "-t",
+                sub_env_path,
+                "-r",
+                req_file,
+            ]
+        )
+        sys.path.append(sub_env_path)
+    except:
+        logging.error(
+            f"Failed to install requirement for Competition Agent in folder {Path(policy_path).name}"
+        )
+        raise
+
+    # insert submission path
+    while policy_path in sys.path:
+        sys.path.remove(policy_path)
+
+    sys.path.insert(0, policy_path)
+
+    # import policy module
+    policy_file_path = str(os.path.join(policy_path, "policy.py"))
+    policy_spec = importlib.util.spec_from_file_location(
+        "competition_policy", policy_file_path
+    )
+    policy_module = importlib.util.module_from_spec(policy_spec)
+    sys.modules["competition_policy"] = policy_module
+    if policy_spec:
+        policy_spec.loader.exec_module(policy_module)
+
+    policy = policy_module.Policy()
+    wrappers = policy_module.submitted_wrappers()
+
+    from .competition_agent import CompetitionAgent
+
+    def env_wrapper(env):
+        import gym
+
+        env = gym.Wrapper(env)
+        for wrapper in wrappers:
+            env = wrapper(env)
+
+        return env
+
+    config = load_config(Path(os.path.join(policy_path, "config.yaml")))
+
+    spec = AgentSpec(
+        interface=resolve_agent_interface(
+            img_meters=int(config["img_meters"]),
+            img_pixels=int(config["img_pixels"]),
+            action_space="TargetPose",
+        ),
+        agent_params={
+            "policy_path": policy_path,
+            "policy": policy,
+        },
+        adapt_env=env_wrapper,
+        agent_builder=CompetitionAgent,
+    )
+
+    # delete competition policy module and remove related path
+    while policy_path in sys.path:
+        sys.path.remove(policy_path)  # prevent duplicate path remain in sys.path
+
+    while sub_env_path in sys.path:
+        sys.path.remove(sub_env_path)  # prevent duplicate path remain in sys.path
+
+    # remove all modules related to policy_path
+    for key, module in list(sys.modules.items()):
+        if "__file__" in dir(module):
+            module_path = module.__file__
+            if module_path and (
+                policy_path in module_path or sub_env_path in module_path
+            ):
+                sys.modules.pop(key)
+
+    del policy_module
```

Comment:
Why are the modules removed at this point?

Reply:
If the user doesn't called competition_agent.close(), this will ensure they don't have import issues when they import the next competition_agent.

Reply:
Then one final question: Is the policy module not needed anymore such that the object can be deleted?

Reply:
I think after the policy & submitted wrapper is once been stored by a variable then the module is not needed anymore. Although keeping the policy_module may not affect the import of next agent.